src/main/java/org/wso2/carbon/inbound/custom/poll/MQPollingInbound.java

```
@@ -0,0 +1,221 @@
+            if (this.injectingSeq == null || "".equals(this.injectingSeq)) {
+                logger.error("Sequence name not specified. Sequence : " + this.injectingSeq + " for " + name);
+                return false;
+            }
+            SequenceMediator seq = (SequenceMediator) this.synapseEnvironment.getSynapseConfiguration()
+                    .getSequence(this.injectingSeq);
+            seq.setErrorHandler(this.onErrorSeq);
+            if (seq != null) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("injecting message to sequence : " + this.injectingSeq + " of " + name);
+                }
+                if (!this.synapseEnvironment.injectInbound(msgCtx, seq, this.sequential)) {
+                    return false;
+                }
+            } else {
+                logger.error("Sequence: " + this.injectingSeq + " not found for " + name);
+            }
+
+        } catch (Exception e) {
+            throw new SynapseException("Error while processing the IBM MQ Message ", e);
+        }
+        return true;
+    }
+
+    /**
+     * Create the message context.
+     */
+    private MessageContext createMessageContext() {
+        MessageContext msgCtx = this.synapseEnvironment.createMessageContext();
+        org.apache.axis2.context.MessageContext axis2MsgCtx = ((Axis2MessageContext) msgCtx).getAxis2MessageContext();
+        axis2MsgCtx.setServerSide(true);
+        axis2MsgCtx.setMessageID(UUID.randomUUID().toString());
+        return msgCtx;
+    }
+
+    /**
+     * Stopping the inbound endpoint
+     */
+    public void destroy() {
+
+        logger.info("Removing resources");
+        try {
+            if (manager.isConnected()) {
+                manager.close();
+                manager = null;
+            } else {
+                manager = null;
+            }
+        } catch (MQException e) {
+            logger.error("Error removing services");
```

Exception is dropped from the log error. Please include exception in log.
logger.error("Error removing services", e);